lib/src/parser/parse.dart

```diff
@@ -193,26 +200,43 @@ class Parser {
             start: startIndex,
             end: endIndex,
           );
-          if (!_excel._sheetMap[sheetName]!._spanList.contains(spanObj)) {
-            _excel._sheetMap[sheetName]!._spanList.add(spanObj);
+          if (!sheet._spanList.contains(spanObj)) {
+            sheet._spanList.add(spanObj);
+
+            _deleteAllButTopLeftCellsOfSpanObj(spanObj, sheet);
           }
           _excel._mergeChangeLookup = sheetName;
         }
       });
     });
+  }
 
-    // Remove those cells which are present inside the
-    _excel._sheetMap.forEach((sheetName, sheetObject) {
```

Comment:
This code was never executed, hence the error it contains never occurred. When executed properly, it would have triggered a ConcurrentModificationError. This is due to the impossibility of removing items from maps while iterating over them using forEach.

Instead, this functionality was implemented in `_deleteAllButTopLeftCellsOfSpanObj`. This approach is also more performant as it avoids iterating over every sheet, every row, and every cell. Instead, it only iterates over the actual ranges of the cell spans.